app/frontend/Calendar/EditEvent/DialogHeader.svelte

```diff
@@ -118,14 +202,74 @@
     onClose();
   }
 
+  async function onChangeAll() {
+    if (!confirmAndChangeRule()) {
+      return;
+    }
+    let master = event.parentEvent;
```

Comment:
Please move the non-UI part of this function into logic/ .

----

In general:
* JS in the UI (frontend, Svelte) should be only be about 1-2 lines. Any function more complex than that should be in logic/ .
* However, logic MUST NOT contain UI access.

Please try to factor the functions so to make both rules true.